design/general_refactorings.md

```diff
@@ -0,0 +1,49 @@
+Before refactor concrete thing here is a list of things that need to be done all over the code:
+
+1. Go Over All Namings And Improve them
```

Comment:
1. Improve naming

I would also detail what do you mean by naming. Is it class naming? object naming? all of the above? Add them in (). For example (class naming, object naming)

I would also provide a short sentence of how this will help us... what is the benefit?